source/auth/auth.rst

```diff
@@ -683,6 +683,176 @@ mechanism
+X-MongoDB-Server-Nonce*  Base64 string of server nonce
+X-MongoDB-GS2-CB-Flag*   ASCII lower-case character ‘n’ or ‘y’ or ‘p’
+X-MongoDB-Optional-Data* Optional, Optional data, base64 encoded representation of the optional object provided by the client
+Body                     Action=GetCallerIdentity&Version=2011-06-15\n
+======================== ======================================================================================================
+
+.. note::
+        ``*``, Denotes a header that MUST be included in SignedHeaders if present.
+
+.. note::
+        Region is not a header, but simply part of the authorization header. Region by default is ‘us-east-1’ since this is the implicit region for ‘sts.amazonaws.com’.
+        Drivers will need to derive the region to use from the endpoint. The region is the second piece of a DNS name. While all official AWS STS 
+        endpoints start with “sts.”, there are non-AWS hosted endpoints and test endpoints that will not follow this rule.
+
+================= =========
+Host              Region
+================= =========
+sts.amazonaws.com us-east-1
+first.second      second
+first             us-east-1
+================= ========= 
+
+Example:
+
+.. code:: javascript
+
+           Content-Length:43
+           Content-Type:application/x-www-form-urlencoded
+           Host:sts.amazonaws.com
+           X-Amz-Date:20191017T173547Z
+           X-MongoDB-Server-Nonce:enJwWTtNSkR+WztFZCE3d1NWSiMpfU54YCgmPU5lY1RHbnN1IWy6vp7GvmtRmcGWYEtjedGEI0ZXi13r7y4V+A==
+           X-MongoDB-GS2-CB-Flag:n
+           Authorization:AWS4-HMAC-SHA256 Credential=AKIAICGVLKOKZVY3X3DA/20191017/us-east-1/sts/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date;x-mongodb-gs2-cb-flag;x-mongodb-server-nonce, Signature=6f67c4dfbc05da04ea8d571bc6f7d11833f9d1909d0068be302c5379e1867d15
+           Action=GetCallerIdentity&Version=2011-06-15
+
+
+`MongoCredential`_ Properties
+`````````````````````````````
+
+username
+        MAY be specified. The non-sensitive IAM access key.
+
+source
+	MUST be "$external". Defaults to ``$external``.
+
+password
+	MAY be specified. The sensitive IAM secret key.
+
+mechanism
+	MUST be "MONGODB-IAM"
+
+mechanism_properties
+	AWS_SESSION_TOKEN
+		Drivers MUST allow the user to specify an AWS session token for authentication with temporary credentials.
+
+IAM Credentials
+
+If a username and password is provided drivers MUST use these for the IAM access key and IAM secret key, respectively. Drivers
+MUST use this secret key to generate a signature using HMAC-SHA256.
+
+Temporary Credentials
+
+If a username and password is not provided drivers MUST query the standard local AWS endpoint for temporary credentials.
+If temporary credentials cannot be obtained then drivers MUST fail authentication and raise an error.
+
+EC2 Instance
```

Comment:
Looks like this is a sub-section of the Temporary Credentials section? If so, can we add some formatting to make that clearer?

Reply:
Done

Reply:
@prashantmital please comment back or resolve.